### ISSUE TITLE
Rule Proposal: Pascal to local and kebab to global

### ISSUE THREAD

**Issue by vhoyer** [vuejs/eslint-plugin-vue#1207]
**Please describe what the rule should do:**
More like an alteration of `component-name-in-template-casing`, I'd like to enforce the use of Pascal case name components on local registered components (to easily find them on the file by using a simple find) and enforce kebab-case on global components, to easily distinguish between the two types.

**What category should the rule belong to?**

- [x] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```js
import Vue from 'vue'
import MyComponent from 'path'
import GlobalComponent from 'path'

Vue.component('global-component', GlobalComponent)
export default {
  components: {
    MyComponent,
  },
  template: `
    <div>
      <MyComponent /> <!-- this is good -->
      <global-component /> <!-- this is good -->

      <my-component /> <!-- this is bad -->
      <GlobalComponent /> <!-- this is bad -->
    </div>
  `
}
```

**Additional context**

**Comment by ota-meshi** [comment]
Thank you for this proposal.

I think it's technically impossible to implement your rules. It is not possible to know from each `.vue` file whether the component was registered globally. So I close this issue.
Thank you for your understanding.

**Comment by vhoyer** [comment]
but you can see which components are defined locally, through the `components` object. This logic even exists on `vue/component-name-in-template-casing`

**Comment by ota-meshi** [comment]
We know that the explicitly defined local component is a local component, but we don't know if the other components are globally defined components. It may have been registered using a mixin.
